CMake/ClangCompilers.cmake

```diff
@@ -104,8 +104,16 @@ set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -ffast-math")
 set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -ffast-math")
 
 # Set extra debug flags
-set(CMAKE_C_FLAGS_DEBUG "${CMAKE_C_FLAGS_DEBUG} -fno-omit-frame-pointer -fstandalone-debug")
-set(CMAKE_CXX_FLAGS_DEBUG "${CMAKE_CXX_FLAGS_DEBUG} -fno-omit-frame-pointer -fstandalone-debug")
+set(CMAKE_C_FLAGS_DEBUG "${CMAKE_C_FLAGS_DEBUG} -fno-omit-frame-pointer")
+set(CMAKE_CXX_FLAGS_DEBUG "${CMAKE_CXX_FLAGS_DEBUG} -fno-omit-frame-pointer")
+
+# unfortunately this removes standalone-debug altogether for offload builds
+# but until we discover how to use the ${OPENMP_OFFLOAD_COMPILE_OPTIONS} more selectively
+# this is the only way to avoid a warning per compilation unit that contains an omp symbol.
+if (NOT OFFLOAD_TARGET MATCHES "nvptx64")
+  set(CMAKE_C_FLAGS_DEBUG "${CMAKE_C_FLAGS_DEBUG} -fstandalone-debug")
+  set(CMAKE_CXX_FLAGS_DEBUG "${CMAKE_CXX_FLAGS_DEBUG} -fstandalone-debug")
```

Comment:
It think it would be helpful to print a warning/reminder about this workaround for debug builds.

Reply:
Instead of warning when this flag is not added, I'd like to just have a regular message saying this option is added or not. Based on the clang doc https://clang.llvm.org/docs/UsersManual.html, this option turns off the optimizations that reduce the size of debug information in the binary. Since this option is not enabled by default on Linux, a message makes it clear if we are engaging this option in QMCPACK. CMake warnings are usually for things that user really need to pay attention. This one doesn't deserve a warning and a regular message should be sufficient.

Reply:
Since it is only turned on for Debug builds and RelWithDebInfo I don't see the issue.  My experience is without it you're going to have a real mess in the debugger if you link to many libraries not built as part of the project and the debugger can't find debug info (very common if linking to system libraries or release build libraries).

Reply:
If when I am tasked to deal with our AMD issues I will look at AMD.

Reply:
I have been looking at the recent work by the offload target team on llvm in Phabricator and the way they are added the AMD offload is inconsistent with how they did CUDA.

I can say for CUDA this is not the only tooling that got broken in this work. When the offload "path" is triggered in clang I can also no longer parse the abstract syntax trees of any file using rtags or the ast-dump tools that are part of llvm.  This is because they broke part of the llvm driver CUDA API. Although I think they did have it right at first and then forgot about it trying to satisfy the fairly low quality reviews their work on this triggered.

Reply:
The -fstandalone-debug flag seems to be accepted by clang on AMD and aompcc and doesn't issue a warning.

Reply:
thanks mark

Reply:
@markdewing you mean when offloading is enabled? The issue is from OpenMP offload.

Reply:
Yes, when offloading is enabled.